### ISSUE TITLE
Mempool failing to conenct to the database

### ISSUE THREAD

**Issue by hugodopradofernandes** [mynodebtc/mynode#675]
**Describe the bug**
Mempool failing to conenct to the database. Mempool UI shows "offline" status. See logs below
Started after Mynode 0.2.57 update. Mempool 2.4.0
Already tried the docker restart/cleanup

**MyNode hardware (please complete the following information):**
 - Device:  raspi4 32-bit 4Gb
 - Version  0.2.57

LOG: 
```
-- Logs begin at Mon 2022-06-27 11:31:47 -03, end at Mon 2022-06-27 11:57:36 -03. --
Jun 27 11:57:31 myNode mempool[18999]: api_1  | Jun 27 14:57:31 [81] INFO: Connected to Electrum Server at 172.26.0.1:50001 (["electrs/0.9.3","1.4"])
Jun 27 11:57:31 myNode mempool[18999]: api_1  |       PRIMARY KEY (hei' at line 12
Jun 27 11:57:31 myNode mempool[18999]: api_1  |       median_fee double unsigned NOT NULL,
Jun 27 11:57:31 myNode mempool[18999]: api_1  | Jun 27 14:57:31 [81] ERR: MIGRATIONS: Unable to create required tables, aborting in 10 seconds. Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'json NOT NULL,
Jun 27 11:57:30 myNode mempool[18999]: api_1  | Jun 27 14:57:30 [81] INFO: Database connection established.
Jun 27 11:57:30 myNode mempool[18999]: api_1  | Jun 27 14:57:30 [81] NOTICE: Starting Mempool Server... (?)
Jun 27 11:57:30 myNode mempool[18999]: api_1  | git not found, cannot parse git hash
Jun 27 11:57:28 myNode mempool[18999]: api_1  | wait-for-it.sh: db:3306 is available after 0 seconds
Jun 27 11:57:28 myNode mempool[18999]: api_1  | wait-for-it.sh: waiting 720 seconds for db:3306
Jun 27 11:57:27 myNode mempool[18999]: mempool_api_1 exited with code 255
```

**Comment by tehelsper** [comment]
I think this is an issue with the 32-bit image. You could upgrade to 64-bit or wait for a fix. I will look into it.

**Comment by erebor** [comment]
Is there a decent guide to upgrading a running node to 64-bit image without, you know, shooting myself in the foot?  Got a running lightning node on mine that I'd prefer not to nuke.

**Comment by tehelsper** [comment]
It's not very risky, especially with this a secondary SD card.

- Get new 32GB+ SD
- Flash 64-bit image onto new SD card
- Shut down device
- Swap SD cards
- Boot back up

**Comment by john-zaprite** [comment]
It's really simple. This is the procedure I followed and all went fine:
1. Shut down node and remove SD card
2. Flash the new image onto the SD (I didn't delete the card first. I just flashed the new image straight on) 
3. Insert the SD back into the Pi and reboot

**Comment by erebor** [comment]
Does the config stuff all stay safely around?  Like, it just comes back up running like it was, but now happily 64-bit?  Cuuuuuuz...that would be most excellent

**Comment by tehelsper** [comment]
Nearly everything should. Your password will be reset to bolt and any optional apps will need to be re-installed. Installing them will get back to the same state. Data *may* be lost for Dojo and BTC Pay server since 32->64 does reset docker data. It's meant to be as seamless as possible.

**Comment by john-zaprite** [comment]
Yeah if you have a 2nd SD I would use that, but it all seemed to work fine for me with the same SD.
Some settings were lost, for example:
- password was reset
- dark mode was reset to light
- some apps had to be reinstalled 

But all balances and channels etc. were all intact. I don't use the node heavily, so I can't say if any other data was lost or not, but for me it all seems fine so far.

**Comment by erebor** [comment]
thanks all!  I guess as long as I keep the original SD handy I can always swap it back in and boot up, yeah?

**Comment by hugodopradofernandes** [comment]
I did that, on 64bit version the external ssd keeps disconnecting and the message "please connect the external disk" appears 😕

**Comment by john-zaprite** [comment]
I think that happened me after the upgrade now that you mention it. But I just rebooted a couple of times and it found the drive and has been fine ever since.

**Comment by tehelsper** [comment]
Check the SSD adapter as well. It's rare, but some drivers are not as well supported in the 64-bit OS. Maybe try the UGREEN adapter suggested on mynodebtc.com/download.
